stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -1003,8 +1061,60 @@ public class MessageListViewModel(
          * @param user The user to be blocked.
          * @param cid The full channel id, i.e. "messaging:123".
          */
+        @Deprecated(
+            "Deprecated in order to bring make the action more explicit." +
+                "Use `MessageListViewModel.ShadowBanUser` if you want to keep the same functionality " +
+                "(shadow banning) or `MessageListViewModel.BanUser` if you want to ban a user.",
+            replaceWith = ReplaceWith(
+                "ShadowBanUser",
+                "package com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel.ShadowBanUser"
+            )
+        )
         public data class BlockUser(val user: User, val cid: String) : Event()
```

Comment:
The new ban/ unban, shadow ban/ remove shadow ban `Event`s are more in line with the API and more explicit.
It's implemented in order to give the user more direct control.